### PR TITLE
Fix listing of tables when schema is not provided for Sheets connector

### LINKED ISSUE #5271: SHEETS_UNKNOWN_SCHEMA_ERROR while accessing google sheets information_schema
tldr; Getting this error while accessing google sheets information_schema

```
SHEETS_UNKNOWN_SCHEMA_ERROR: Schema not present - Optional.empty
```

We recently upgraded our installation to Presto 340 and since then I'm not able to query inside the information_schema for all the google sheets schemas I have.
```
$ presto --catalog=sheets --schema=information_schema

presto:information_schema> select * from tables;
Query 20200922_190402_07566_25xyh, FAILED, 2 nodes
Splits: 17 total, 0 done (0.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]
Query 20200922_190402_07566_25xyh failed: Schema not present - Optional.empty
presto:information_schema>
```
Surprisingly, adding a limit/where at the end of the query works
```
presto:information_schema> select * from tables limit 100;
 table_catalog |    table_schema    |           table_name           | table_type
---------------+--------------------+--------------------------------+------------
 risksheets    | information_schema | columns                        | BASE TABLE
 risksheets    | information_schema | tables                         | BASE TABLE
 risksheets    | information_schema | views                          | BASE TABLE
 risksheets    | information_schema | schemata                       | BASE TABLE
 risksheets    | information_schema | table_privileges               | BASE TABLE
 risksheets    | information_schema | roles                          | BASE TABLE
 risksheets    | information_schema | applicable_roles               | BASE TABLE
 risksheets    | information_schema | enabled_roles                  | BASE TABLE
 risksheets    | information_schema | role_authorization_descriptors | BASE TABLE
 risksheets    | default            | metadata_table                 | BASE TABLE
 risksheets    | default            | sample_table1                  | BASE TABLE
 risksheets    | default            | sample_table2                  | BASE TABLE
(12 rows)

Query 20200922_194223_00447_r2nkd, FINISHED, 2 nodes
Splits: 17 total, 17 done (100.00%)
0:00 [12 rows, 809B] [40 rows/s, 2.66KB/s]

presto:information_schema> select * from tables where table_schema='default';
 table_catalog | table_schema |   table_name   | table_type
---------------+--------------+----------------+------------
 risksheets    | default      | metadata_table | BASE TABLE
 risksheets    | default      | sample_table1  | BASE TABLE
 risksheets    | default      | sample_table2  | BASE TABLE
(3 rows)

Query 20200922_195026_00616_r2nkd, FINISHED, 2 nodes
Splits: 17 total, 17 done (100.00%)
0:01 [3 rows, 181B] [4 rows/s, 262B/s]

presto:information_schema>
```
And these are from the server logs
```
2020-09-22T19:04:02.219Z	ERROR	SplitRunner-124-127182	io.prestosql.execution.executor.TaskExecutor	Error processing Split 20200922_190402_07566_25xyh.1.0-0 io.prestosql.connector.informationschema.InformationSchemaSplit@58517a03 (start = 5.0174100655561E7, wall = 0 ms, cpu = 0 ms, wait = 0 ms, calls = 1): SHEETS_UNKNOWN_SCHEMA_ERROR: Schema not present - Optional.empty
io.prestosql.spi.PrestoException: Schema not present - Optional.empty
	at io.prestosql.plugin.google.sheets.SheetsMetadata.listTables(SheetsMetadata.java:140)
	at io.prestosql.metadata.MetadataManager.listTables(MetadataManager.java:571)
	at io.prestosql.metadata.MetadataListing.listTables(MetadataListing.java:93)
	at io.prestosql.connector.informationschema.InformationSchemaPageSource.addTablesRecords(InformationSchemaPageSource.java:281)
	at io.prestosql.connector.informationschema.InformationSchemaPageSource.buildPages(InformationSchemaPageSource.java:219)
	at io.prestosql.connector.informationschema.InformationSchemaPageSource.getNextPage(InformationSchemaPageSource.java:183)
	at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:301)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1076)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_33d3221____20200922_061905_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Some additional observations:
1. I can issue show tables on information_schema but can't query any.
2. I can query all the tables in the default schema (the one with the actual data of google sheets) just fine.

Any ideas?

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #5271

### CODE REVIEW

File: presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java [hunk 1]
```diff
@@ -62,6 +62,8 @@ public void testListTable()
     {
         assertQuery("show tables", "SELECT * FROM (VALUES 'metadata_table', 'number_text', 'table_with_duplicate_and_missing_column_names')");
         assertQueryReturnsEmptyResult("SHOW TABLES IN gsheets.information_schema LIKE 'number_text'");
+        assertQuery("select table_name from gsheets.information_schema.tables WHERE table_schema <> 'information_schema'", "SELECT * FROM (VALUES 'metadata_table', 'number_text', 'table_with_duplicate_and_missing_column_names')");
+        assertQuery("select table_name from gsheets.information_schema.tables WHERE table_schema <> 'information_schema' LIMIT 1000", "SELECT * FROM (VALUES 'metadata_table', 'number_text', 'table_with_duplicate_and_missing_column_names')");
```
Comment (findepi) [inline presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java:66]: would we be able to run `io.prestosql.testing.AbstractTestIntegrationSmokeTest#testSelectInformationSchemaTables` and `testSelectInformationSchemaColumns` for Google Sheets?

Comment (Praveen2112) [reply presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java:66]: Once we add support for writes in sheets connector we will be able to run these tests. Or if we could push the tpch data into the sheets which is specified in the test then we could run these tests.

File: presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java [hunk 2]
```diff
@@ -136,16 +133,15 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     @Override
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
     {
-        if (schemaName.isEmpty()) {
-            throw new PrestoException(SHEETS_UNKNOWN_SCHEMA_ERROR, "Schema not present - " + schemaName);
-        }
-        if (schemaName.get().equalsIgnoreCase("information_schema")) {
-            return ImmutableList.of();
+        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
```
Comment (phd3) [inline presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java:136]: my understanding is that currently the only "default" schema is supported (apart from information schema). The "sheetsClient" doesn't seem to understand the concept of a "schema", but it is being repeatedly invoked in a loop iterating over schema objects. So there's kind of an assumption that SCHEMAS has size=1. 

I feel we could make the logic more explicit by:

- either making `sheetsClient#getTableNames` aware of schema names
- or having `SheetsMetadata#listTables` method explicitly check that "default" is the desired schema, before invoking `sheetsClient#getTableNames()`. 

Thoughts?

**Review by phd3 — COMMENTED** [review]
Thanks for investigating and fixing this! :)

**Review by phd3 — APPROVED** [review]
Looks good, just one suggestion.

**Pull request merged**